### DISCUSSION TITLE
Throwing melees & `boomerang` Testing & Feedback

### DISCUSSION THREAD

**Discussion by MXKhronos** [HelixNebulaStudio/RiseOfTheDead#505] [General]
## New throwing mechanic
Made throwing melees more intuitive, the thrown melee should generally land where your crosshair is aiming at, the closer the target, the more likely it will land accurately, the accuracy is based on they thrown `velocity + velocityBonus * throwCharge`, which aren't intended to be upgradable values.

**Throw Stamina Cost**
This cost comes with the increase of `ThrowDamagePercent` of the throwable melees, from ~2% to 4% overall. (Is that too much or suffice?) All are subject to change.

## Boomerang
The boomerang is a throwing focused pointed melee, it's designed to be short melee range but more throwable.

Need some balancing feedback for it.
- Melee: Base Damage, Stamina Cost, Attack Speed..
- Throw: Throw Rate, Stamina Cost

Animations are still work in progress.

**Comment by MrPigZ** [comment]
I can't view broom spear or boomerang in workbench
![image](https://github.com/HelixNebulaStudio/RiseOfTheDead/assets/162422862/a9a6e638-1029-4e90-abce-63701a03a98f)

**Comment by Leroy8888** [comment]
I can't seem to view the `velocity`, `velocityBonus` or `throwCharge` stats on the weapons in the workbench. Maybe the distance of the thrown weapon could potentially increase the longer the player charges the weapon? It currently doesn't seem to do that and instead just acts as a cooldown between thrown attacks.

**Comment by MXKhronos** [comment]
> I can't seem to view the `velocity`, `velocityBonus` or `throwCharge` stats on the weapons in the workbench. Maybe the distance of the thrown weapon could potentially increase the longer the player charges the weapon? It currently doesn't seem to do that and instead just acts as a cooldown between thrown attacks.

Currently the velocity affects the time it takes to reach your aiming target and the max distance before it cuts off from attempting to reach your target point, meaning it will try to reach it but it will run out of force.

I'll add those stats in to workbench.

**Comment by choppygraph** [comment]
I really like how the boomerang feels, it's satisfying to use against longer range enemies and surprisingly accurate when thrown. At close quarters however the combat feels finicky, but if it's meant for longer ranges I can see why that would be so. 

Testing it with regular enemies was fine, however it does kinda shred bosses with its regular close range attack. It managed to rival my vector X when going against the tanker, and felt like I was carrying an infinite Molotov while fighting Robert.

Stamina cost for throwing seems perfect, the stamina cost for swinging felt a little low.
For my final thoughts on it, it feels a little too good right now.

**Comment by MrGarrettFerret** [comment]
The boomerang feels amazing to use, especially with the new updates to throwing melees.

The weapon feels fairly balanced except for a couple things, the melee attack speed and stamina cost. I do feel as if the melee attack speed should be slowed down a little, maybe to 0.45 or 0.5. The stamina cost for melee attacks should also be increased a bit, 4 is way too low for the amount of damage it can output. Maybe increase it to 8 or 9?

**Comment by MXKhronos** [comment]
- Reduced attack speed to 0.45
- Attack Stamina cost to 8

Adjusted Velocities and Velocity Bonus, throw charge bonuses effects are more noticeable now.

**Comment by Leroy8888** [comment]
I was able to see a slight increase in range when I charged some of the throwable melees, but it still feels rather lackluster. It's almost something you wouldn't pay notice to since the charge time is insanely short on all weapons, especially for the boomerang with just 0.2 seconds.

I would personally like to suggest to increase the charge time on all weapons to a minimum of 2 seconds while also increasing the charge velocity bonus. But instead of having to fully charge the weapon to gain the bonus, the bonus would depend on the charge meter. This is similar to how the focus mechanic with damage of the tactical bow used to work. This would allow players to adjust the amount of time they charge depending on the distance of the target while retraining a constant DPS (closer = more DPS since less charge time).

It also feels like the throw damage percentage on the pickaxe is a bit low for the amount of skill it needs to put it into use. It's a rather heavy tool with a lot of mass and you can definitely feel that when you throw it. It is obtainable at a rather low level, but regardless of that I think a buff to it from 4% to 5 or 6% would be justified as there already seems to be a cap at 50k for any entity.

**Comment by Leroy8888** [comment]
The boomerang in terms of stats seems fine at the moment. Maybe a slight extra nerf on the attack speed, but that also kind of depends on how heavy the usage between thrown melee and regular melee attacks is prefered to be distributed on the tier 3 melee. In my own opinion I do find 51k dps (with melee fury and Swordman skill) still relatively high for a weapon meant to be mostly thrown.

If you would compare this weapon to the other throwable ones like the broomspear and pickaxe how heavily would the boomerang rely on the throw aspect? Or is it supposed to be an additional tool at the disposal of a melee weapon on top of the stats it has?

The ricochet of the throw is a bit unreliable and seems to hit 2 targets at most usually. It also has a tendency to get stuck on objects, entities or floors.

It also seems to have an issue with perspectives. When used in first person the melee attack is faced to the left of the character while in third person the attack faces forwards. But that would likely be be fixed if it receives new animations.

**Comment by choppygraph** [comment]
Are there any mod ideas in particular unique to pointed weapons planned?

**Comment by Leroy8888** [comment]
Will the snowball also receive some changes to it or will it be kept the same?

**Comment by MXKhronos** [comment]
> Are there any mod ideas in particular unique to pointed weapons planned?

Not yet, am open to suggestions.


> Will the snowball also receive some changes to it or will it be kept the same?

Snowballs are currently a different class of weapons but they will also eventually be updated with the new throwing mechanics and ironing out the kinks.
